process/block/metablock.go

```diff
@@ -1838,6 +1867,20 @@ func (mp *metaProcessor) checkShardHeadersFinality(
 			continue
 		}
 
+		if mp.enableEpochsHandler.IsFlagEnabledInEpoch(common.EquivalentMessagesFlag, lastVerifiedHdr.GetEpoch()) {
```

Comment:
does it fits better to check proof in `checkShardHeadersValidity` which is called just before instead of here?

Reply:
I think it should stay here, as here is the place where we check for finality